### PR TITLE
add dnsimple provider

### PR DESCRIPTION
**Pull request by nlf**
also tweaked the init function in the route53 handler so it doesn't cause the whole thing to error out if you're not using it

### CODE REVIEW

File: providers/route53.go [hunk 1]
```diff
@@ -21,6 +21,21 @@ var (
 )
 
 func init() {
+	if len(os.Getenv("AWS_REGION")) == 0 {
```
Comment (alena1108) [inline providers/route53.go]: Nit: as we are evaluating all env vars here, the check below is no longer needed:

https://github.com/rancher/external-dns/blob/a76ffa3d3804e926c4399504bf368311a4566e9a/providers/route53.go#L44

Comment (nlf) [inline providers/route53.go]: true. i erred on the side of modifying as little as possible to make this work, but you're right that those checks can be removed now.

**Comment by alena1108** [comment]
@nlf LGTM. Once that extra check is removed, we can merge it. Let me know if you could remove this check, or I can do it.

**Comment by nlf** [comment]
done

**Comment by vincent99** [comment]
@nlf you'll want to add an entry based on https://github.com/rancher/rancher-catalog/tree/master/templates/route53 too so it will show up in the catalog.

@lvuch can you make a catalog icon for it?  Here's there logo:

DNSimple

**Comment by alena1108** [comment]
@nlf new image  rancher/external-dns:v0.1.8 having dnsimple provider, is pushed to dockerhub. Let me know if you face any problems using it.

File: providers/dnsimple.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+		recordTTLs[fqdn] = map[string]int{}
+		recordTTLs[fqdn][rec.Type] = rec.TTL
+		recordSet, exists := recordMap[fqdn]
+		if exists {
+			recordSlice, sliceExists := recordSet[rec.Type]
+			if sliceExists {
+				recordSlice = append(recordSlice, rec.Content)
+			} else {
+				recordSet[rec.Type] = []string{rec.Content}
+			}
```
Comment (janeczku) [inline providers/dnsimple.go]: @nlf Nice code but you are missing a line here: The `recordSlice` is never written back to the map and therefore all additional records get lost and are not returned. Maybe do a follow up?
```GO
			if sliceExists {
				recordSlice = append(recordSlice, rec.Content)
				recordSet[rec.Type] = recordSlice
			} else {
```

Comment (nlf) [inline providers/dnsimple.go]: oh dang, you're right. good catch. i'll push a fix.

**Pull request merged**